diff --git a/internal/client/index.js b/internal/client/index.js
index a7e08d94d947308bdf78d9a116c3b296ae46b00f..ec91e66fdea3b7b208d22ef2302fa0739df03db3 100644
--- a/internal/client/index.js
+++ b/internal/client/index.js
@@ -180,7 +180,7 @@ var FORM_ASSOCIATED_CUSTOM_ELEMENT_CALLBACKS = [
 // src/client/client-window.ts
 import { BUILD as BUILD6 } from "@stencil/core/internal/app-data";
 var win = typeof window !== "undefined" ? window : {};
-var doc = win.document || { head: {} };
+var doc = win.document || { head: {}, querySelectorAll: () => [] };
 var H = win.HTMLElement || class {
 };
 var plt = {
